### DISCUSSION TITLE
creating tables

### DISCUSSION THREAD

**Discussion by aryan02420** [cronvel/terminal-kit#186] [Q&A] [answered]
What is the difference between `.createInlineElement(TextTable)` and `.table()`?

```js
import termkit from 'terminal-kit'
const term = termkit.terminal

const tableRows = [
  ['Column1', 'Column2', 'Column3'],
  ['Row1', 'Row1', 'Row1'],
  ['Row2', 'Row2', 'Row2'],
  ['Row3', 'Row3', 'Row3'],
]

const tableProps = {
  fit: false,
}

term.createInlineElement(termkit.TextTable, {          // <-- how is this
  cellContents: [...tableRows],
  ...tableProps,
})

term.table([...tableRows], { ...tableProps })          // <-- different from this?
```

Both of them produce the exact same output. Do they differ only in the default value of `fit` property?

Also, is it possible to set table height to auto? The tables always have padding on the bottom to fit the initial size of the terminal, but i want to have the second table to begin right after the first one ends. Also when the terminal is smaller than the table height, the tables get cropped of from the bottom. I am probably missing something.

![image](https://user-images.githubusercontent.com/58023300/134969051-2f181291-d47f-481f-a3d1-b45703add7ab.png)
![image](https://user-images.githubusercontent.com/58023300/134969838-a48a8522-0358-4c08-994f-c073b86ff0ab.png)

Thanks, 
aryan

**Comment by cronvel** [comment]
Hi @aryan02420 

Basically, there is no difference.

Now for the *why*: this lib has two modes, the first one is straight-forward and low-level and is called the inline mode, the second one is called the *document model* mode where you define a page with widgets (useful for terminal-based application).
`term.TextTable` is a widget of the *document model*, while `term.table()` is the *inline mode* counter-part.
By the way, to avoid coding things twice, many widgets can be used inline, using `term.createInlineElement()`, and that's why `term.table()` is using `term.createInlineElement()` behind the scene, probably with some default options.

**Comment by cronvel** [comment]
@aryan02420 the blank line is a bug, I wonder when it has appeared, I'm on it.

About the crop when the terminal is too small, it will take much longer to fix. Like I said, it used a document model's widget, and those widgets are rendered only on the terminal area (as if it was a screen), it doesn't scroll.

**Comment by cronvel** [comment, marked as answer]
@aryan02420 Ok it's fixed as of v2.1.7.
The crop bug was not an issue after all (I had already done things for supporting that).

Thanks for reporting ;)
